research-software-directory/RSD-as-a-service
# 986 reference papers

# Add reference papers to software edit section

Closes #289 
Closes #756 
Closes #986 

Changes proposed in this pull request:
* The user can add the reference papers used to cite their software.
* The scraper will search for all citations using OpenAlex api.
* List all the citations and which reference paper is cited in the "Referenced by" tab

How to test:
* `make start` to build app, after app is started you will need to stop it and start it with scrapers. You can use `docker compose down && docker compose up -d`
* login as rsd-admin in order to be able to edit existing software, otherwise you can create new software items
* select first software from the list, and edit the software.
* rename software to `CDK reference test` so you can find it easily later, 
* navigate to reference papers section
* review `no items to show` messages and provide improvement suggestions (if any)
* add following CDK reference papers by DOI
    *  doi:10.1186/s13321-017-0220-4
    *  10.1186/1758-2946-6-3
    *  https://doi.org/10.1021/ci025584y
    * doi:10.1021/ci025584y
* after you added reference papers you should wait ca. 10 min for scrapers to start working or run manually 
```bash
docker compose exec scrapers java -cp /usr/myjava/scrapers.jar nl.esciencecenter.rsd.scraper.doi.MainCitations
```
* in the meantime navigate to some other pages. When you get back to reference papers page the count in the tab should show the number of citations scraped
* take another software or create one and try some other reference papers. AFAIK any DOI paper will be scraped for the citations (it does not have to be related to a software at all).

## Start screen reference papers page
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/99fec22e-d487-41c5-ad1c-f74825a5149f)

![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/96c55484-9537-4e3c-9327-0c1a5847a424)

## Added reference papers
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/7b05aa4e-05db-4f00-abf4-ee39901758c7)

## Scraper found some citations
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/6a4a51b8-873d-4152-a9e6-afba6a831485)


PR Checklist:
